sql_extensions/migrations/00002_latest_periodic_state.sql

```diff
@@ -0,0 +1,227 @@
+    ON (registration.data -> 'market_metadata' ->> 'market_id')::NUMERIC = volume.market_id
+LEFT JOIN inbox_latest_periodic_state AS periodic_state
+    ON (registration.data -> 'market_metadata' ->> 'market_id')::NUMERIC = periodic_state.market_id;
+-- noqa: disable=ST06
+
+CREATE INDEX inbox_latest_periodic_state_by_tvl ON inbox_events (
+    ((data ->> 'tvl_per_lp_coin_growth_q64')::NUMERIC) DESC
+) WHERE (data ->> 'ends_in_bonding_curve')::BOOLEAN = true;
+
+-- }}}
+
+-- ADD LOCK TO PREVIOUS TRIGGERS {{{
+
+CREATE OR REPLACE FUNCTION UPDATE_LATEST_STATE()
+RETURNS TRIGGER AS $$
+BEGIN
+  PERFORM pg_advisory_xact_lock(42);
+  INSERT INTO inbox_latest_state (
+    sequence_number,
+    creation_number,
+    account_address,
+    transaction_version,
+    transaction_block_height,
+    type,
+    data,
+    inserted_at,
+    event_index,
+    indexed_type,
+    event_name,
+    market_id
+  )
+  SELECT
+    NEW.sequence_number,
+    NEW.creation_number,
+    NEW.account_address,
+    NEW.transaction_version,
+    NEW.transaction_block_height,
+    NEW.type,
+    NEW.data,
+    NEW.inserted_at,
+    NEW.event_index,
+    NEW.indexed_type,
+    NEW.event_name,
+    (NEW.data -> 'market_metadata' ->> 'market_id')::NUMERIC
+  ON CONFLICT (market_id) DO UPDATE SET
+    transaction_version = EXCLUDED.transaction_version,
+    transaction_block_height = EXCLUDED.transaction_block_height,
+    data = EXCLUDED.data,
+    inserted_at = EXCLUDED.inserted_at,
+    event_index = EXCLUDED.event_index
+  WHERE
+    (inbox_latest_state.data -> 'market_metadata' ->> 'market_id')::NUMERIC = (EXCLUDED.data -> 'market_metadata' ->> 'market_id')::NUMERIC
+  AND
+    (inbox_latest_state.data -> 'state_metadata' ->> 'market_nonce')::NUMERIC < (EXCLUDED.data -> 'state_metadata' ->> 'market_nonce')::NUMERIC;
+  RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE OR REPLACE FUNCTION UPDATE_VOLUME()
+RETURNS TRIGGER AS $$
+BEGIN
+  PERFORM pg_advisory_xact_lock(42);
+  INSERT INTO inbox_volume (market_id, all_time_volume, volume_events)
+  VALUES (
+    (NEW.data->'market_metadata'->>'market_id')::numeric,
+    0::numeric,
+    '[]'::jsonb
+  )
+    ON CONFLICT (market_id) DO NOTHING;
+  UPDATE inbox_volume
+  SET
+    volume_events = (
+      SELECT COALESCE(
+        jsonb_agg(jsonb_build_object('time', (e.data->'periodic_state_metadata'->>'start_time')::numeric, 'volume_quote', (e.data->>'volume_quote')::numeric)),
+        '[]'::jsonb
+      )
+      FROM inbox_events e
+      WHERE e.event_name = 'emojicoin_dot_fun::PeriodicState'
+        AND (e.data->'periodic_state_metadata'->>'start_time')::numeric / 1000000 > extract(epoch from (now() - interval '1 day'))
+        AND e.data->'periodic_state_metadata'->>'period' = '60000000'
+        AND (e.data->'market_metadata'->>'market_id')::numeric = (NEW.data->'market_metadata'->>'market_id')::numeric
+    ),
+    all_time_volume = inbox_volume.all_time_volume + (NEW.data->>'volume_quote')::numeric
```

Comment:
This method is not as accurate as calculating all_time_volume from `State.cumulative_stats.quote_volume`

With this method, volume will only update at the end of a minute

Why not get `all_time_volume` directly from `State.cumulative_stats.quote_volume`?